Lib/pathlib.py

```diff
@@ -119,10 +120,10 @@ def _select_from(self,  parent_path, scandir):
 
 class _WildcardSelector(_Selector):
 
-    def __init__(self, pat, child_parts, flavour):
-        flags = re.NOFLAG if _is_case_sensitive(flavour) else re.IGNORECASE
+    def __init__(self, pat, child_parts, case_sensitive):
+        flags = re.NOFLAG if case_sensitive else re.IGNORECASE
```

Comment:
The selector classes are internal APIs, so when we come to add directory-by-directory case sensitivity, we can freely move the `case_sensitive` argument from `__init__()` to `_select_from()` without anyone noticing.

But for the time being, case sensitivity when `case_sensitive=None` depends only on the flavour, and not the path. I feel the current implementation should reflect that, rather than preparing for future work that may end up blocked for one reason or another. Maybe?

Reply:
It'll mainly just undo the changes you've done here, is the problem. We'd need to plumb the flavour back through to the selector instance as well as the case sensitivity preference, and probably put the function onto the flavour itself so that they can provide their own default.

Changing it to take the path can wait, but given that WSL directories when accessed from Windows are already case sensitive, I'm surprised we haven't had more people asking for better support. I guess we probably don't have many places that need it.

Reply:
Gotcha, thanks for explaining. I'll undo the change to the selector initialiser arguments.

Reply:
I've restored the `flavour` argument. I'm still passing `case_sensitive` to the initialiser rather than `_select_from()` for performance's sake - hope that's what you had in mind?